apps/web/src/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer.tsx

```diff
@@ -42,92 +66,187 @@ export const AutocompleteItem = (item: { tokenInfo: TokenInfo; balance: string }
+  const canAddMoreRecipients = useMemo(() => recipientFields.length < MAX_RECIPIENTS, [recipientFields])
 
-  const selectedToken = balancesItems.find((item) => item.tokenInfo.address === tokenAddress)
-  const { totalAmount, spendingLimitAmount } = useTokenAmount(selectedToken)
+  const addRecipient = (): void => {
+    if (!canAddMoreRecipients) {
+      setCsvAirdropModalOpen(true)
+      return
+    }
 
-  const canCreateSpendingLimitTxWithToken = useHasPermission(Permission.CreateSpendingLimitTransaction, {
-    token: selectedToken?.tokenInfo,
-  })
+    if (recipientFields.length === 1) {
+      setMaxRecipientsInfo(true)
+    }
 
-  const isSpendingLimitType = type === TokenTransferType.spendingLimit
+    append({
+      recipient: '',
+      tokenAddress: ZERO_ADDRESS,
+      amount: '',
+    })
+  }
 
-  const maxAmount = isSpendingLimitType && totalAmount > spendingLimitAmount ? spendingLimitAmount : totalAmount
+  const removeRecipient = (index: number): void => {
+    if (recipientFields.length > 1) {
+      remove(index)
+    }
+  }
 
-  const isAddressValid = !!recipient && !errors[TokenTransferFields.recipient]
+  const csvAirdropAppUrl = safeApps?.[0]?.url
 
-  useEffect(() => {
-    setNonceNeeded(!isSpendingLimitType || spendingLimitAmount === 0n)
-  }, [setNonceNeeded, isSpendingLimitType, spendingLimitAmount])
+  const CsvAirdropLink = () => (
+    <Link sx={{ cursor: 'pointer' }} onClick={() => setCsvAirdropModalOpen(true)}>
+      CSV Airdrop
+    </Link>
+  )
+
+  const canBatch = isMassPayoutsEnabled && type === TokenTransferType.multiSig
 
   return (
     <TxCard>
       <FormProvider {...formMethods}>
         <form onSubmit={handleSubmit(onSubmit)} className={commonCss.form}>
-          <FormControl fullWidth sx={{ mt: 1 }}>
-            <AddressBookInput name={TokenTransferFields.recipient} canAdd={isAddressValid} />
-          </FormControl>
+          <Stack spacing={3}>
+            <Stack spacing={8}>
+              {recipientFields.map((field, index) => (
+                <RecipientRow
+                  key={field.id}
+                  removable={recipientFields.length > 1}
+                  fieldArray={{ name: MultiTokenTransferFields.recipients, index }}
+                  remove={removeRecipient}
+                  disableSpendingLimit={disableSpendingLimit || recipientFields.length > 1}
+                />
+              ))}
+            </Stack>
+
+            {canBatch && (
+              <>
+                <Stack direction="row" alignItems="center" justifyContent="space-between" mb={4}>
+                  <Button
+                    data-testid="add-recipient-btn"
+                    variant="text"
+                    onClick={addRecipient}
+                    disabled={!canAddMoreRecipients}
+                    startIcon={<SvgIcon component={AddIcon} inheritViewBox fontSize="small" />}
+                    size="large"
+                  >
+                    Add Recipient
```

Comment:
```suggestion
                    Add recipient
```

Here and everywhere else: please don't use Cap Case in the UI.

@IvaLukan please update the designs accordingly. Thanks!

Reply:
Fixed